hardware/EnOceanESP3.cpp

```diff
@@ -4139,3 +4140,41 @@ int getPositionFromCommandLevel(int cmnd , int pos )
 
 	return pos;
 }
+void CEnOceanESP3::DeleteSensor(const std::string& sensorId)
+{
+		m_sql.safe_exec_no_return("DELETE FROM EnoceanSensors WHERE (DeviceID == '%q')", sensorId.c_str() );
+        _log.Debug(DEBUG_NORM, "CSQLHelper::DeleteDevices: EnoceanSensors  ID: %s", sensorId.c_str());
+        LoadNodesFromDatabase();
+}
+//delete the sensor in EnoceanSensors table assossiated to the device with ID in deviceStatus table
+//take care that we could have many device.
+void EnOceanDeleteSensorAssociatedWithDevice(const std::string& ID)
+{
+		//get the number of enocean devices with same DeviceId in order to delete from EnoceanSensors
+		std::string SensorDeviceID="";
+		std::vector<std::vector<std::string> > result;
+        //get DeviceID = DeviceID in EnoceanSensors
+		result = m_sql.safe_query("SELECT DeviceID , HardwareID  FROM DeviceStatus   WHERE (ID==%s) ", (ID).c_str());
+		if (result.size()>0)	{
+            SensorDeviceID   = result[0][0];
+			std::string HwID = result[0][1];
+
+            //check if it is ESP3  hardware
+            CEnOceanESP3 *pEnoceanHardware = reinterpret_cast<CEnOceanESP3 *>( m_mainworker.GetHardwareByIDType(HwID, HTYPE_EnOceanESP3));
```

Comment:
Do not repeat the type
```suggestion
            auto EnoceanHardware = reinterpret_cast<CEnOceanESP3 *>( m_mainworker.GetHardwareByIDType(HwID, HTYPE_EnOceanESP3));
```

Reply:
Thx for the suggestions. Could be useful if maybe you can also educate us by telling why 😁

Reply:
> Thx for the suggestions. Could be useful if maybe you can also educate us by telling why

Sure ! C++ has evolved quite a lot since the 80's and it should looks more and more like Python nowadays. ;-) 
A good starting point: https://isocpp.org/ and more specifically about my comments: http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines like in http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-auto about avoiding repeating the type when it is obvious or non interesting.